cmd/cluster/core/create.go

```diff
@@ -1159,3 +1192,269 @@ func validateVersion(ctx context.Context, versionCLI string, client crclient.Cli
+		return nil, fmt.Errorf("etcd sharding config file must contain at least one shard")
+	}
+
+	return config.Shards, nil
+}
+
+// parseInlineShards parses multiple inline shard definitions from --etcd-shard flags.
+func parseInlineShards(shardDefs []string) ([]hyperv1.ManagedEtcdShardSpec, error) {
+	shards := make([]hyperv1.ManagedEtcdShardSpec, 0, len(shardDefs))
+
+	for i, def := range shardDefs {
+		shard, err := parseInlineShard(def)
+		if err != nil {
+			return nil, fmt.Errorf("invalid shard definition at index %d: %w", i, err)
+		}
+		shards = append(shards, shard)
+	}
+
+	return shards, nil
+}
+
+// parseInlineShard parses a single inline shard definition.
+// Format: name=<name>,prefixes=<prefix1>|<prefix2>,priority=<priority>,replicas=<n>,storage-size=<size>,storage-class=<class>,backup-schedule=<schedule>
+func parseInlineShard(def string) (hyperv1.ManagedEtcdShardSpec, error) {
+	shard := hyperv1.ManagedEtcdShardSpec{}
+	pairs := strings.Split(def, ",")
+
+	var hasName, hasPrefixes bool
+
+	for _, pair := range pairs {
+		kv := strings.SplitN(pair, "=", 2)
+		if len(kv) != 2 {
+			return shard, fmt.Errorf("invalid key=value pair: %s", pair)
+		}
+
+		key := strings.TrimSpace(kv[0])
+		value := strings.TrimSpace(kv[1])
+
+		switch key {
+		case "name":
+			shard.Name = value
+			hasName = true
+		case "prefixes":
+			shard.ResourcePrefixes = strings.Split(value, "|")
+			hasPrefixes = true
+		case "priority":
+			shard.Priority = hyperv1.EtcdShardPriority(value)
+		case "replicas":
+			replicas, err := strconv.ParseInt(value, 10, 32)
+			if err != nil {
+				return shard, fmt.Errorf("invalid replicas value: %w", err)
+			}
+			r := int32(replicas)
+			shard.Replicas = &r
+		case "storage-size":
+			if shard.Storage == nil {
+				shard.Storage = &hyperv1.ManagedEtcdStorageSpec{
+					Type:             hyperv1.PersistentVolumeEtcdStorage,
+					PersistentVolume: &hyperv1.PersistentVolumeEtcdStorageSpec{},
+				}
+			}
+			size, err := resource.ParseQuantity(value)
+			if err != nil {
+				return shard, fmt.Errorf("invalid storage-size value: %w", err)
+			}
+			shard.Storage.PersistentVolume.Size = &size
+		case "storage-class":
+			if shard.Storage == nil {
+				shard.Storage = &hyperv1.ManagedEtcdStorageSpec{
+					Type:             hyperv1.PersistentVolumeEtcdStorage,
+					PersistentVolume: &hyperv1.PersistentVolumeEtcdStorageSpec{},
+				}
+			}
+			shard.Storage.PersistentVolume.StorageClassName = ptr.To(value)
+		case "backup-schedule":
+			shard.BackupSchedule = ptr.To(value)
+		default:
+			return shard, fmt.Errorf("unknown key: %s", key)
+		}
+	}
+
+	if !hasName {
+		return shard, fmt.Errorf("name is required")
+	}
+	if !hasPrefixes {
+		return shard, fmt.Errorf("prefixes is required")
+	}
+
+	return shard, nil
+}
+
+// applyGlobalDefaults applies global etcd configuration defaults to shards that don't have specific values.
+func applyGlobalDefaults(shards []hyperv1.ManagedEtcdShardSpec, opts *CreateOptions) {
```

Comment:
This function name doesn't feel like it will age well, not linked to etcd shards in any way